### ISSUE TITLE
[CLI] config reset

### ISSUE THREAD

**Issue by daro1337** [scramjetorg/transform-hub#394]
**Feature description**
I had some problems with incorrect configuration and after running e.g. `si -h` would throw an error and the only way to fix it was to manually remove `~/.sirc /.sth-cli-rc.json`

Suggestion:
`si config reset`

We should ask user if he is sure he wants to delete the configuration

**Comment by alicja-gruzdz** [comment]
Interesting, I received such errors few times but for me it happened only when I installed malfunctioning CLI. I wonder what happened in your case.
In the https://github.com/scramjetorg/transform-hub/pull/364 the `config reset` was introduced. However, from what I'm checking it does not reset all values in the config but only the selected ones:

![Zrzut ekranu z 2022-04-05 09-36-51](https://user-images.githubusercontent.com/8177865/161703375-20258598-89f2-41d2-a8aa-c77a49e8402a.png)

However, I understand what you propose is something simpler and more bullet proof which is deleting the existing global and session configuration files. @ErykSol what do you think about it? Of course not for now, but for the future CLI enrichment.

**Comment by f1ames** [comment]
> Interesting, I received such errors few times but for me it happened only when I installed malfunctioning CLI. I wonder what happened in your case.

Hmmm, it happened to me when I had `middlewareApiUrl` set and `production` env like:

```
  "middlewareApiUrl": "http://",
  "env": "production",
```

Any `si` command (even `si config ...`) tried to request MW, and as it was unreachable threw:

```
Error: Failed to request 'http://127.0.0.1:7000/api/v1/spaces' with code ECONNREFUSED.
```

So resetting `si` configuration manually seemed like the only option (and it helped). @daro1337 was it similar in your case?

**Comment by alicja-gruzdz** [comment]
That's a great feedback, thanks @f1ames We'll test it while we're at SCP CLI dev.

**Comment by ErykSol** [comment]
In [#364](https://github.com/scramjetorg/transform-hub/pull/364) `si config reset all` will cause same results as removing configuration file. Also `~/.sirc /.sth-cli-rc.json` will be moved to new location `~/.si/global-config.json`.

@f1ames in [#364] we've added configuration validation for CLI, so invalid urls format shouldn't be possible anymore, although we will test more for invalid middleware urls, to investigate it further. Thanks for feedback.

**Comment by ErykSol** [comment]
Fixed in #403.
